client_encryption/json_path_utils.py

```diff
@@ -42,7 +42,12 @@ def update_node(tree, path, node_str):
         except json.JSONDecodeError:
             node_json = node_str
 
-        if to_set in current_node and type(current_node[to_set]) is dict and type(node_json) is dict:
```

Comment:
To avoid: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&branch=fix_array_payload&id=Mastercard_client-encryption-python&open=AYMOH13rUj1Lk7_J0HyB

You could refactor like: 
 ```
if to_set in current_node and (type(current_node[to_set]) is dict or type(current_node) is list) and (type(node_json) is dict or type(current_node) is list):
```
and add this check before:
```
        if type(current_node) is list:
            current_node = current_node[0]
```